service-loadbalancer/service_loadbalancer.go

```diff
@@ -352,6 +356,12 @@ func (lbc *loadBalancerController) getEndpoints(
 // - :80 services don't need a :80 postfix
 // - default ns should be accessible without /ns/name (when we have /ns support)
 func getServiceNameForLBRule(s *api.Service, servicePort int) string {
+	if *nameLabel != "" {
```

Comment:
This is actually one of the reasons we need Ingress, so we can let users choose urls to their Services, but I'm OK just allowing this. However, I think you should put the new string into the service Annotations like we did with a couple of other options (https://github.com/kubernetes/contrib/blob/master/service-loadbalancer/service_loadbalancer.go#L52). then you can also set/get it via the serviceAnnotations type @aledbf put in.